### ISSUE TITLE
[AppBar] Catalog examples cannot tap Bar Buttons

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#2096]
### Overview
The AppBar examples (except for Storyboard) have non-tappable buttons in the NavBar/Flexible Header view.  If you scroll the AppBar so that it goes up (toward the Status Bar), the buttons become tappable.

Some views have the navigation elements in the wrong position.

![simulator screen shot - iphone se - 2017-10-06 at 02 51 52](https://user-images.githubusercontent.com/1753199/31242297-7ff45aca-aa41-11e7-99aa-730b61252a37.png)

### Reproduction steps
1. Open MDC Catalog &rarr; App Bar &rarr; Demo
2. Tap on the Back button in the App Bar

#### Expected Result
View Controller is popped

#### Actual Result
No effect

### Version number
35.2.0

### Operating system and device
iPhone SE/iOS 11.0 (Simulator), Xcode 9 (GM)

**Comment by romoore** [comment]
FYI, this is develop/bff04f190d8b3685b5e1036913f26d79f545be06

**Comment by chuga** [comment]
I haven't been able to repro, can you clarify the steps?

![simulator screen shot - iphone 6 - 2017-10-05 at 14 12 10](https://user-images.githubusercontent.com/4545451/31243107-338a33a4-a9d7-11e7-99e6-a6a57a88b00a.png)

**Comment by romoore** [comment]
Quicktime not uploadable?? ( ò_Ó )
![app_bar-bug](https://user-images.githubusercontent.com/1753199/31245071-f2983f08-aa44-11e7-8dbe-1d09215c4c0d.gif)

**Comment by romoore** [comment]
May be related to a broken UIConstraint ("Demo" example)

```
2017-10-06 03:20:37.008749+0900 MDCCatalog[87474:2094542] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x6080000982e0 MDCHeaderStackView:0x7fcae66602e0.top == UILayoutGuide:0x6080001ae1c0'UIViewSafeAreaLayoutGuide'.top   (active)>",
    "<NSLayoutConstraint:0x60800009ed20 MDCHeaderStackView:0x7fcae66602e0.bottom == UIView:0x7fcae6660e00.bottom   (active)>",
    "<NSLayoutConstraint:0x60800029ea50 'UIView-Encapsulated-Layout-Height' UIView:0x7fcae6660e00.height == 76   (active)>",
    "<NSLayoutConstraint:0x608000096cb0 'UIViewSafeAreaLayoutGuide-top' V:|-(96)-[UILayoutGuide:0x6080001ae1c0'UIViewSafeAreaLayoutGuide']   (active, names: '|':UIView:0x7fcae6660e00 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60800009ed20 MDCHeaderStackView:0x7fcae66602e0.bottom == UIView:0x7fcae6660e00.bottom   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
```

**Comment by romoore** [comment]
Could not reproduce on iPhone 5c/iOS 10.3.3, but could reproduce on iPhone 5s/iOS 9.3 (Simulator) from XCode 9 (GM).  May be a simulator bug from Xcode 9 (GM).

**Comment by chuga** [comment]
Yeah, I was able to repro on Xcode 9 / iOS 11.0, but things work as expected on 9.1 / 11.1.
I'll test on a device with 11.0 and report back.

**Comment by chuga** [comment]
Ok, this is a bug on iOS 11.0 that appears to be fixed on iOS 11.1. And the PR where this was introduced is https://github.com/material-components/material-components-ios/pull/2078.
😖
